### PR TITLE
Move cleric doctrine data to enable homebrew

### PR DESCRIPTION
**Pull request by TikaelSol**
Migration number is 831 to allow for #6640 to be merged prior to this one more painlessly.

### CODE REVIEW

**Comment by stwlam** [comment]
How does this and the barbarian changes relate?

**Comment by TikaelSol** [comment]
Just anticipating that one getting merged first so the conflict with this one is smaller.

File: static/lang/re-en.json [hunk 1]
```diff
@@ -1248,6 +1248,7 @@
             },
             "Cleric": {
                 "Doctrine": {
+                    "Drops": "1st-level Cleric class feature",
```
Comment (stwlam) [inline static/lang/re-en.json:1251]:
```suggestion
                    "AllowedDrops": "1st-level cleric class feature",
```

File: packs/data/classfeatures.db/doctrine.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
             {
                 "adjustName": false,
                 "allowedDrops": {
-                    "label": "1st-level Cleric class feature",
+                    "label": "PF2E.SpecificRule.Cleric.Doctrine.Drops",
```
Comment (stwlam) [inline packs/data/classfeatures.db/doctrine.json:28]:
```suggestion
                    "label": "PF2E.SpecificRule.Cleric.Doctrine.AllowedDrops",
```

File: src/module/migration/migrations/831-cleric-doctrines.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                secondDoctrine: "Compendium.pf2e.classfeatures.sa7BWfnyCswAvBVa",
+                thirdDoctrine: "Compendium.pf2e.classfeatures.s8WEmc4GGZSHSC7q",
+                fourthDoctrine: "Compendium.pf2e.classfeatures.vxOf4LXZcqUG3P7a",
+                fifthDoctrine: "Compendium.pf2e.classfeatures.n9W8MjjRgPpUTvWf",
+                finalDoctrine: "Compendium.pf2e.classfeatures.DgGefatQ4v6xT6f9",
+            },
+        };
+    }
+
+    get #warpriestSetFlags(): AELikeSource {
```
Comment (stwlam) [inline src/module/migration/migrations/831-cleric-doctrines.ts:28]:
```suggestion
    get #warpriestFlags(): AELikeSource {
```

File: src/module/migration/migrations/831-cleric-doctrines.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import { RuleElementSource } from "@module/rules";
+import { MigrationBase } from "../base";
+
+/**
+ * Update rule elements on the Cleric doctrines to include references to the granted doctrines.
+ */
+export class Migration831clericDoctrines extends MigrationBase {
+    static override version = 0.831;
+
+    get #cloisteredClericSetFlags(): AELikeSource {
```
Comment (stwlam) [inline src/module/migration/migrations/831-cleric-doctrines.ts:12]:
```suggestion
    get #cloisteredClericFlags(): AELikeSource {
```

File: src/module/migration/migrations/831-cleric-doctrines.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            return;
+        }
+
+        switch (source.system.slug) {
+            case "cloistered-cleric": {
+                source.system.rules.push(this.#cloisteredClericSetFlags);
+                break;
+            }
+            case "warpriest": {
+                source.system.rules.push(this.#warpriestSetFlags);
```
Comment (stwlam) [inline src/module/migration/migrations/831-cleric-doctrines.ts:62]:
```suggestion
                source.system.rules.push(this.#warpriestFlags);
```

File: src/module/migration/migrations/831-cleric-doctrines.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                (r: MaybeAELikeSource): r is MaybeAELikeSource =>
+                    r.key === "ActiveEffectLike" && r.path === "flags.pf2e.cleric"
+            )
+        ) {
+            return;
+        }
+
+        switch (source.system.slug) {
+            case "cloistered-cleric": {
+                source.system.rules.push(this.#cloisteredClericSetFlags);
```
Comment (stwlam) [inline src/module/migration/migrations/831-cleric-doctrines.ts:58]:
```suggestion
                source.system.rules.push(this.#cloisteredClericFlags);
```

File: src/module/migration/runner/base.ts [hunk 7]
```diff
@@ -14,7 +14,7 @@ interface CollectionDiff<T extends foundry.data.ActiveEffectSource | ItemSourceP
 export class MigrationRunnerBase {
     migrations: MigrationBase[];
 
-    static LATEST_SCHEMA_VERSION = 0.829;
+    static LATEST_SCHEMA_VERSION = 0.831;
```
Comment (stwlam) [inline src/module/migration/runner/base.ts:17]: This going to break migrations for anyone running dev mode. Change this to 0.83 and the barbarian one to 0.831.

**Review by stwlam — CHANGES_REQUESTED** [review]

File: static/lang/re-en.json (hunk 1 above)
Comment (stwlam) [reply static/lang/re-en.json]: Change all the rest if you want to break convention and capitalize this one. :)

File: src/module/migration/runner/base.ts (hunk 7 above)
Comment (TikaelSol) [reply src/module/migration/runner/base.ts:17]: So let's get the barbarian one merged first and I can fix the conflicts here.

File: static/lang/re-en.json [hunk 8]
```diff
@@ -1263,6 +1263,7 @@
             },
             "Cleric": {
                 "Doctrine": {
+                    "AllowedDrops": "1st-level Cleric class feature",
```
Comment (TikaelSol) [inline static/lang/re-en.json:1266]:
```suggestion
                    "AllowedDrops": "1st-level cleric class feature",
```

**Review by stwlam — APPROVED** [review]

**Pull request merged**